modules/phase_field/include/kernels/MatGradSqCoupled.h

```diff
@@ -0,0 +1,40 @@
+#ifndef MATGRADSQCOUPLED_H
+#define MATGRADSQCOUPLED_H
+
+#include "Kernel.h"
+#include "JvarMapInterface.h"
+#include "DerivativeMaterialInterface.h"
+
+// Forward Declarations
+class MatGradSqCoupled;
+
+template <>
+InputParameters validParams<MatGradSqCoupled>();
+
+/*
+ * This kernel calculates the prefactor * nabla_psi term in A-C equation for phase field modeling of oxidation
```

Comment:
I suggest removing `prefactor * nabla_psi` here since you give the form of the term below as `prefactor * grad_psi * grad_psi`. Also please change A-C -> Allen-Cahn